Restore program options for RViz

botteroa-si opened the pull request:
closes #223 

This pull request restores the program options that can currently be used.
In particular, the Ogre output is now hidden by default and can be enabled via the corresponding option.

botteroa-si committed:
Restore working program options

greimela-si committed:
Hide ogre output and enable with program option

botteroa-si committed:
Fix uncrustify and Windows build

botteroa-si commented:
CI:

* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=4281)](http://ci.ros2.org/job/ci_linux/4281/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=1290)](http://ci.ros2.org/job/ci_linux-aarch64/1290/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=3518)](http://ci.ros2.org/job/ci_osx/3518/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=4364)](http://ci.ros2.org/job/ci_windows/4364/)

greimela-si committed:
Move ros initialization after command line parsing

This allows the help option to exit correctly.

greimela-si committed:
Pass correct program args in visual testing

botteroa-si commented:
@wjwwood, the instability of the Windows build seems to be unrelated to our changes.

Martin-Idel-SI committed:
Delete verbose option as it is now unsupported

Martin-Idel-SI commented:
New CI after rebase:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=4418)](http://ci.ros2.org/job/ci_linux/4418/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=1366)](http://ci.ros2.org/job/ci_linux-aarch64/1366/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=3613)](http://ci.ros2.org/job/ci_osx/3613/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=4483)](http://ci.ros2.org/job/ci_windows/4483/)

Martin-Idel-SI commented:
New CI after rebase:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=4428)](http://ci.ros2.org/job/ci_linux/4428/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=1374)](http://ci.ros2.org/job/ci_linux-aarch64/1374/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=3619)](http://ci.ros2.org/job/ci_osx/3619/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=4491)](http://ci.ros2.org/job/ci_windows/4491/)

wjwwood commented on rviz_common/src/rviz_common/visualizer_app.cpp:
```diff
@@ -147,108 +150,110 @@ bool VisualizerApp::init(int argc, char ** argv)
+//       ros::console::notifyLoggerLevelsChanged();
+//     }
+//   }
+  } catch (std::exception & e) {
```
It might be nice to have a more specific try-catch here, maybe just whatever Qt throws if it fails to parse correctly. Could be done in a follow up pr though.

botteroa-si replied:
It actually turned out that the method `QCommandLineParser::process()` takes care internally of the exception handling, so that no try-catch block is needed. Therefore we just removed it.

wjwwood commented on rviz_visual_testing_framework/src/visual_test_fixture.cpp:
```diff
@@ -38,13 +38,17 @@
   QLocale::setDefault(QLocale::English);
-  int argc = 0;
+  int argc = 1;
+  char * argv[] = {const_cast<char *>("rviz2"), nullptr};
```
This is scary, since `QApplication()` might try to modify it.

botteroa-si replied:
We have removed the const_cast and now use directly a char array.

wjwwood reviewed (changes requested):
lgtm, just one change in a test where you're casting away const for a literal.

botteroa-si committed:
Remove const_cast

botteroa-si committed:
Remove try-catch block when parsing command line

botteroa-si committed:
Fix GNU warning

botteroa-si committed:
Don't use strcpy() anymore

botteroa-si commented:
@wjwwood, thank you for the comments. We implemented your requested changes and rebased on the new master. 

New CI: 

* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=4465)](http://ci.ros2.org/job/ci_linux/4465/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=1396)](http://ci.ros2.org/job/ci_linux-aarch64/1396/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=3646)](http://ci.ros2.org/job/ci_osx/3646/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=4531)](http://ci.ros2.org/job/ci_windows/4531/)

wjwwood commented:
I restarted CI for you (it looks like a disconnect on our end):

- macOS:
  - [![Build Status](https://ci.ros2.org/buildStatus/icon?job=ci_osx&build=3650)](https://ci.ros2.org/job/ci_osx/3650/)
